src/newtype.rs

```diff
@@ -0,0 +1,57 @@
+use crate::constants::*;
+use rusqlite as sql;
+use rusqlite::types::*;
+use std::fmt;
+
+macro_rules! defnewtype {
+    ($name:ident ($base:ty); $( $field:ident => $value:expr),*) => {
+        #[repr(transparent)]
+        #[derive(Copy, Clone, Eq, PartialEq, Debug)]
+        pub struct $name { value: $base }
+
+        // Display instance reuses instance of underlying type, since it seems
+        // to be impossible to convert identifiers to strings at
```

Comment:
Maybe specify string representations directly, like: `IMAGE => DC_MSG_IMAGE, "Image",`
Macros can parse such definitions.

Reply:
I'd rather not. I expect `defnewtype!` macro be reused, but not all its uses need  conversion to string.